service/collector/bridge.go

```diff
@@ -0,0 +1,164 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"regexp"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	subsystemBridge = "bridge"
+)
+
+var (
+	bridgeClusterWithoutNetworkInterfaceDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystemBridge, "cluster_without_network_interface"),
+		"Clusters without network interface.",
+		[]string{
+			labelCluster,
+		},
+		nil,
+	)
+	bridgeNetworkInterfaceWithoutClusterDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystemBridge, "network_interface_without_cluster"),
+		"Network interfaces without cluster.",
+		[]string{
+			labelCluster,
+		},
+		nil,
+	)
+)
+
+type BridgeConfig struct {
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
+}
+
+type Bridge struct {
+	g8sClient versioned.Interface
+	logger    micrologger.Logger
+}
+
+func NewBridge(config BridgeConfig) (*Bridge, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	c := &Bridge{
+		g8sClient: config.G8sClient,
+		logger:    config.Logger,
+	}
+
+	return c, nil
+}
+
+func (c *Bridge) Collect(ch chan<- prometheus.Metric) error {
+	ctx := context.Background()
+
+	var desiredClusterIDs []string
+	{
+		c.logger.LogCtx(ctx, "level", "debug", "message", "finding all cluster IDs in FlannelConfigs")
+
+		l, err := c.g8sClient.Core().FlannelConfigs(metav1.NamespaceAll).List(metav1.ListOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, i := range l.Items {
+			desiredClusterIDs = append(desiredClusterIDs, i.Name)
+		}
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d cluster IDs: %#v", len(desiredClusterIDs), desiredClusterIDs))
+	}
+
+	var bridgeClusterIDs []string
+	{
+		c.logger.LogCtx(ctx, "level", "debug", "message", "finding network interfaces")
+
+		interfaces, err := net.Interfaces()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, i := range interfaces {
+			id := clusterIDFromName(i.Name)
+			if id == "" {
```

Comment:
This combined with comment below is a bit confusing. Took me some good time to understand why we cannot parse all interfaces. I think this would be more obvious if `clusterIDFromName()` returned an error when parsing fails.

Reply:
Again, the other collector works the same way.